src/SBVonKarman.cpp

```diff
@@ -0,0 +1,395 @@
+    {
+        assert(dynamic_cast<const SBVonKarmanImpl*>(_pimpl.get()));
+        return static_cast<const SBVonKarmanImpl&>(*_pimpl).getLam();
+    }
+
+    double SBVonKarman::getR0() const
+    {
+        assert(dynamic_cast<const SBVonKarmanImpl*>(_pimpl.get()));
+        return static_cast<const SBVonKarmanImpl&>(*_pimpl).getR0();
+    }
+
+    double SBVonKarman::getL0() const
+    {
+        assert(dynamic_cast<const SBVonKarmanImpl*>(_pimpl.get()));
+        return static_cast<const SBVonKarmanImpl&>(*_pimpl).getL0();
+    }
+
+    double SBVonKarman::getScale() const
+    {
+        assert(dynamic_cast<const SBVonKarmanImpl*>(_pimpl.get()));
+        return static_cast<const SBVonKarmanImpl&>(*_pimpl).getScale();
+    }
+
+    bool SBVonKarman::getDoDelta() const
+    {
+        assert(dynamic_cast<const SBVonKarmanImpl*>(_pimpl.get()));
+        return static_cast<const SBVonKarmanImpl&>(*_pimpl).getDoDelta();
+    }
+
+    double SBVonKarman::getDeltaAmplitude() const
+    {
+        assert(dynamic_cast<const SBVonKarmanImpl*>(_pimpl.get()));
+        return static_cast<const SBVonKarmanImpl&>(*_pimpl).getDeltaAmplitude();
+    }
+
+    double SBVonKarman::getHalfLightRadius() const
+    {
+        assert(dynamic_cast<const SBVonKarmanImpl*>(_pimpl.get()));
+        return static_cast<const SBVonKarmanImpl&>(*_pimpl).getHalfLightRadius();
+    }
+
+    double SBVonKarman::structureFunction(double rho) const
+    {
+        assert(dynamic_cast<const SBVonKarmanImpl*>(_pimpl.get()));
+        return static_cast<const SBVonKarmanImpl&>(*_pimpl).structureFunction(rho);
+    }
+
+    //
+    //
+    //
+    //VonKarmanInfo
+    //
+    //
+    //
+
+    const double VonKarmanInfo::magic1 = 2*boost::math::tgamma(11./6)/(pow(2, 5./6)*pow(M_PI, 8./3))
+                                    * pow(24/5.*boost::math::tgamma(6./5), 5./6);
+    const double VonKarmanInfo::magic2 = boost::math::tgamma(5./6)/pow(2., 1./6);
+    const double VonKarmanInfo::magic3 = VonKarmanInfo::magic1*boost::math::tgamma(-5./6)/pow(2., 11./6);
+    const double VonKarmanInfo::magic4 = boost::math::tgamma(11./6)*boost::math::tgamma(5./6)
+                                    / pow(M_PI,8./3)
+                                    * pow(24./5*boost::math::tgamma(6./5),5./6);
+
+    class VKIkValueResid {
+    public:
+        VKIkValueResid(const VonKarmanInfo& vki, double mkt) : _vki(vki), _mkt(mkt) {}
+        double operator()(double k) const {
+            double val = _vki.kValue(k)-_mkt;
+            xdbg<<"resid(k="<<k<<")="<<val<<'\n';
+            return val;
+         }
+    private:
+        const double _mkt;
+        const VonKarmanInfo& _vki;
+    };
+
+    VonKarmanInfo::VonKarmanInfo(double lam, double r0, double L0, bool doDelta,
+                                 const GSParamsPtr& gsparams) :
+        _lam(lam), _r0(r0), _L0(L0), _r0L0m53(pow(r0/L0, -5./3)), _gsparams(gsparams),
+        _deltaAmplitude(exp(-0.5*magic4*_r0L0m53)),
```

Comment:
This is the only place magic4 shows up.  Probably better to just define it right above this function, so the math is visible to anyone coming back to it later.